### ISSUE TITLE
After the latest VSCode automatic update, I am unable to connect to my remote host.

### ISSUE THREAD

**Issue by vartyagi** [microsoft/vscode-remote-release#9478]
Type: <b>Bug</b>

After the latest VSCode update, I am unable to connect to my remote host. When I monitor the Remote SSH logs it keeps printing "Waiting for server log" multiple times before showing the error message "Could not establish connection to <IP address>. The VS Code Server failed to start". 

I even tried the same after deleting my ".vscode-server" directory in my remote home folder but this did not resolve the issue. 

Extension version: 0.108.0
VS Code version: Code 1.86.0 (05047486b6df5eb8d44b2ecd70ea3bdf775fd937, 2024-01-31T10:28:19.990Z)
OS version: Windows_NT x64 10.0.22621
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz (8 x 2112)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.85GB (7.31GB free)|
|Process Argv|--crash-reporter-id 86968068-3783-414e-95c1-61bc51a4aeac|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vscod805:30301674
binariesv615:30325510
vsaa593:30376534
py29gd2263:30899288
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
0bi6i642:30951795
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
dsvsc019bcf:30953938
3ef8e399:30949928

```

</details>

**Comment by vartyagi** [comment]
Digging deeper it seems like their is a new GLIBC requirement:
 Server did not start successfully. Full server log at /home/varun/.vscode-server
> r/.05047486b6df5eb8d44b2ecd70ea3bdf775fd937.log >>>
> Warning: Missing GLIBC >= 2.28! from /lib/x86_64-linux-gnu/libc-2.27.so
> Error: Missing required dependencies. Please refer to our FAQ https://aka.ms/vsc
> code-remote/faq/old-linux for additional information.
> <<< End of server log

Will downgrade to v1.85 as suggested by the docs: 
https://code.visualstudio.com/docs/remote/faq#_can-i-run-vs-code-server-on-older-linux-distributions

**Comment by mtovino-cloudflare** [comment]
I am observing the same behavior.   But also: 

VSCode v1.86 + extension v0.107.1 - works
VSCode v1.85.2 + extension v0.108 - works
VSCode v1.86 + extension v0.108 - does not work

**Comment by DavidC-75** [comment]
> VSCode v1.86 + extension v0.107.1 - works

Not for me...

The Remote-SSH extension deploys on the remote SSH host (linux based) a vscode-server that does not seem to depend on the **extension version**, but rather on the **VS Code version** (visible in Help / about)

* VS Code version 1.86.0:
Remote-SSH deploys vscode-server with commit_id 05047486b6df5eb8d44b2ecd70ea3bdf775fd937 (regardless of the Remote-SSH version) - Requires glibc 2.28

* VS Code Code 1.85.1
Remote-SSH deploys vscode-server with commit_id 8b3775030ed1a69b13e4f4c628c612102e30a681 (regardless of the Remote-SSH version) - Requires glibc 2.22

One would imagine that it's possible to update VS Code (especially if it runs on Windows) **without bringing in new requirements** for the remote SSH host.... But no, a different decision was made. Thanks a lot ;-)

**Comment by mtovino-cloudflare** [comment]
@DavidC-75 interesting... 

... I'm wondering if I'm observing a different failure mode than you are.  Ubuntu 22.04 says: 

```
$ ldd --version
ldd (Ubuntu GLIBC 2.35-0ubuntu3.6) 2.35
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```

Debian 12: 

```
ldd (Debian GLIBC 2.36-9+deb12u4) 2.36
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```

IIUC, both of these _should_ work with 1.86 just fine... and they seem to, but with Remote SSH 0.108... 

... should I file a separate issue?

**Comment by DavidC-75** [comment]
> ... should I file a separate issue?

I guess the answer to that question lies in your `~/.vscode-server/.05047486b6df5eb8d44b2ecd70ea3bdf775fd937.log`

What I'm seeing is
```
Warning: Missing GLIBC >= 2.28! from /lib64/libc-2.22.so
Error: Missing required dependencies. Please refer to our FAQ https://aka.ms/vscode-remote/faq/old-linux for additional information.

```
which leaves no doubt....

**Comment by mtovino-cloudflare** [comment]
Yeap, looks like I need to file a new issue.   

My server-side log when connecting with 1.86+0.108: 

```
Got delay-shutdown request while in shutdown timeout, delaying
[23:06:53] Got delay-shutdown request while in shutdown timeout, delaying
Cancelling previous shutdown timeout
[23:06:53] Cancelling previous shutdown timeout
[23:06:53] [127.0.0.1][2b5bd0b3][ManagementConnection] New connection established.
[23:06:53] [127.0.0.1][b2102798][ExtensionHostConnection] New connection established.
[23:06:53] [127.0.0.1][b2102798][ExtensionHostConnection] <16500> Launched Extension Host Process.
```

**Comment by mtovino-cloudflare** [comment]
I filed https://github.com/microsoft/vscode-remote-release/issues/9479; completely different issue.  

The extension doesn't like comments in an include in my SSH config.  Easily worked around by moving the configs Code needs to know about to a separate config file and then including _that_ from my `~/.ssh/config`.

**Comment by hraywilson** [comment]
Same. I am on CentOS Linux release 7.9.2009 (Core)
I carelessly allowed the VSCode upgrade while clearing a spot for my tea cup and bang ruined my Saturday morning.
Not cool

**Comment by DavidC-75** [comment]
https://github.com/microsoft/vscode-remote-release/issues/9467#issuecomment-1925498025 might help.....

**Comment by hraywilson** [comment]
Thanks David. I'll give it a whirl. Yesterday was a long day with vi.

**Comment by connor4312** [comment]
This is fixed in VS Code 1.86.1

**Comment by Panasavv** [comment]
> This is fixed in VS Code 1.86.1

I am still unable to use Remote-SSH. Last friday i suddenly was unable to connect through VS Code - remote ssh. Then i downgraded VS code to 1.85.2 and remote-ssh to 0.107.1 as suggested here https://github.com/microsoft/vscode-remote-release/issues/9467#issuecomment-1922845326
Everything was fixed and i was able to connect remotely again.

Then today, i was once again unable to establish remote connection with the same issue. I tried uninstalling and reinstalling the same versions (1.85.2 and 0.107.1) but it didn't work. I then saw your message and tried installing 1.86.1 but once again i am not able to connect.

Any help?
